### DISCUSSION TITLE
Install dependencies in prebuild

### DISCUSSION THREAD

**Discussion by buggtb** [loft-sh/devpod#1458] [Q&A] [answered]
hello folks, 

messing with prebuilds and wanted to know, I install yarn / pip etc dependencies as a postCreateCommand which takes time when the instance comes up. Can you call a command in a prebuild to get the deps for you to nullify the need on container startup?

Or are prebuilds everything minus the code in the workspace?

Its not entirely clear in the docs.

**Comment by pascalbreuninger** [comment, marked as answer]
Hey @buggtb, we currently don't include the changes of lifecycle commands like `postCreateCommand` into the prebuild, only features and whatever is in your Dockerfile in case you have one.

This is something we can change though :)

**Reply by buggtb** [reply]
Thanks @pascalbreuninger thanks for the reply.

I'm looking at yarn and poetry dependencies, when the developers create a workspace, on our infra by the time they have done their thing its another 7 or 8 minutes of waiting for the dependencies to download etc. Not your problem I accept, but it is the way software development works these days, for good or bad! :)

I think its worth considering, some way of running lifecycle commands like yarn install so that even though we'd still run it postCreate at that point you've already got 98% of the dependencies downloaded and it perhaps grabs one small update or whatever.
